test_suite/conftest.py

```diff
@@ -32,15 +39,36 @@ def check_markers(host, request):
     ]
 
     host_distro = host.system_info.distribution
-    host_version = host.system_info.release
+    host_version = float(host.system_info.release)
     distro_version = f'{host_distro}{host_version}'
 
     # Skip the test if the distro is explicitly excluded
     if exclude_on_marker:
         exclude_on_marker_list = exclude_on_marker.args[0]
         if host_distro in exclude_on_marker_list \
                 or distro_version in exclude_on_marker_list:
-            pytest.skip(f"This test doesn't apply to {distro_version}")
+            pytest.skip(skip_massage.format(distro_version=distro_version))
+
+        elif len(exclude_on_marker_list) > 1 and host_distro in exclude_on_marker_list[0]:
+            exclude_on_marker_list_version = parse_distro_version(exclude_on_marker_list[0])
+            if exclude_on_marker_list[1] == '<':
+                if host_version < exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version smaller than {exclude_on_marker_list[0]}")
+            elif exclude_on_marker_list[1] == '>':
+                if host_version > exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version bigger than {exclude_on_marker_list[0]}")
+
+    if major_version_marker:
+        major_version_marker_list = major_version_marker.args[0]
+        for major_version in major_version_marker_list:
+            major_version_marker_list_version = parse_distro_version(major_version)
```

Comment:
I find these names a bit confusing, the combination of marker name + list + version makes a bit difficult to follow the code while reading it. It is also because there are other variables named similarly, such as the one above this line: `major_version_marker_list`.

I would suggest to use simpler names and try not to use "version" twice.

Reply:
ACK